libs/utils/src/celix_threads.c

```diff
@@ -236,3 +236,31 @@ celix_status_t celixThreadRwlockAttr_destroy(celix_thread_rwlockattr_t *attr) {
 celix_status_t celixThread_once(celix_thread_once_t *once_control, void (*init_routine)(void)) {
     return pthread_once(once_control, init_routine);
 }
+
+celix_status_t celix_tss_create(celix_tss_key_t* key, void (*destroyFunction)(void*)) {
+    int rc = pthread_key_create(key, destroyFunction);
+    if (rc == 0) {
+        return CELIX_SUCCESS;
+    }
+    return rc == ENOMEM ? CELIX_ENOMEM : CELIX_ILLEGAL_STATE;
```

Comment:
Our error code scheme already contains all errno. 
What about we document it explicitly and use `ENOMEM` (and other errnos) directly without modifying celix_errno.h?